src/main/java/core/levels/LevelManager.java

```diff
@@ -20,18 +22,66 @@
     private final OrthogonalTiledMapRenderer mapRenderer;
     private final WorldManager worldManager;
     private final SpriteBatch batch;
+    private float spawnTime;
 
     public LevelManager(LevelState level) {
         this.level = level;
         map = new TmxMapLoader().load(level.getMapPath());
         this.mapRenderer = new OrthogonalTiledMapRenderer(map, level.getUnitScale());
         this.worldManager = new WorldManager(level.world);
-
         this.batch = new SpriteBatch();
+        this.spawnTime = 15;
     }
 
+    /**
+     * Step physics simulation of World
+     * Elapse time in World
+     * Spawn enemies in World
+     *
+     * Above actions only performed if level isn't in a paused state
+     *
+     * @param dt time delta to simulate (seconds) (capped at .5 in case computer is too slow)
+     */
     public void step(float dt) {
-        worldManager.step(dt);
+        // If level isn't paused, perform step, time elapsing, spawning
+        if (!level.getLevelPaused()) {
+            // Stepping physics simulation
+            worldManager.step(dt);
+
+            // Elapsing time in world
+            level.setCurrentTime(level.getCurrentTime() + dt);
```

Comment:
maybe change this method to `level.incrementTime`, seems cleaner imo

Reply:
wanted to maintain everything as get/set since LevelState an entity class and didn't want any misunderstanding over any method doing anything apart from getting/setting

Reply:
I think it's fine to have increment, imo it's basically just even more restricted setting

up to u tho feel free to ignore and just resolve this